DEVELOPER_GUIDE.md

```diff
@@ -0,0 +1,155 @@
+# Developer Guide
+
+These instructions are intended to help PureScript users more easily contribute to the compiler, even if this is your first Haskell project.
+
+## Prerequisites
+
+Install `stack`. [Instructions](https://docs.haskellstack.org/en/stable/README/).
+
+See the [PureScript Install Guide](INSTALL.md) for information on the `curses`/`ncurses` dependency.
+
+Update `stack` before proceeding:
+```
+stack update
+```
+
+## Select Release
+
+After cloning the PureScript compiler repo, `checkout` the tagged release that you'd like to build.
```

Comment:
If people are trying to contribute, they should almost always be starting from `master`, so I think I'd prefer to remove this bit.

Reply:
Motivations for this section are:
* Letting users profile or fix an issue at a specific release. For example they may encounter new bugs if `master` contains breaking changes.
* Describe how to install a particular release (absorbing the removed INSTALL.md content)

But maybe I'm underestimating the abilities of readers to navigate releases.

Reply:
There's not much point in profiling or fixing an issue at a specific release, though, because it won't be merged if it can't be applied to master, or if the problem no longer exists in master.